### PR TITLE
Filter enrich policy index deletes to just the policy's associated indices

### PR DESCRIPTION
**Pull request by joegallo**
If you have two enrich policies, and the name of the one plus a dash is a prefix for the name of the other (e.g. "policy-test" and "policy-test-more"), then if you delete the prefix policy the indices associated with the other policy will get deleted, too.

### CODE REVIEW

**Comment by joegallo** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Comment by joegallo** [comment]
I did a little bit of spelunking back to when enrich was first added to master via https://github.com/elastic/elasticsearch/pull/48039.

```
    private void prepareAndCreateEnrichIndex() {
         long nowTimestamp = nowSupplier.getAsLong();
         String enrichIndexName = EnrichPolicy.getBaseName(policyName) + "-" + nowTimestamp;
         Settings enrichIndexSettings = Settings.builder()
```

```
     public static String getBaseName(String policyName) {
         return ENRICH_INDEX_NAME_BASE + policyName;
     }
```

```
    public static final String ENRICH_INDEX_NAME_BASE = ".enrich-";
```

The important thing is that it's always been the case that the pattern is `prefix-policy_name-timestamp` -- if that had ever changed, then there'd maybe be some concern here that I was making it possible that some backing indices would never get deleted (like, say, if the pattern had at one point been `prefix-policy_name-BACKING-timestamp` -- the current code would have still deleted that, but my new code would not).

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java [hunk 1]
```diff
@@ -82,12 +83,13 @@ protected void masterOperation(
         ClusterState state,
         ActionListener<AcknowledgedResponse> listener
     ) throws Exception {
-        EnrichPolicy policy = EnrichStore.getPolicy(request.getName(), state); // ensure the policy exists first
+        String policyName = request.getName();
+        EnrichPolicy policy = EnrichStore.getPolicy(policyName, state); // ensure the policy exists first
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java:87]: maybe make these variables explicitly immutable?

**Review by martijnvg — APPROVED** [review]
LGTM, good catch!

**Comment by joegallo** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java (hunk 1 above)
Comment (joegallo) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java]: ✅, d1658cf

**Comment by hendry-lim** [comment]
This PR has not been backported yet.

**Comment by joegallo** [comment]
@hendry-lim hasn't it? I'm seeing bae4e115332ab10c7565c88192bca506a14660a2 for 8.0 and 6bab4774245b55843e27ad457b27cf37a5078278 for 7.17.

**Pull request merged**